event.go

```diff
@@ -96,6 +121,81 @@ func formatHeaders(h http.Header, redact map[string]bool) map[string]string {
 	return ret
 }
 
+func redactValues(b any, includedKeys map[string]bool) any {
+	lowerCase := regexp.MustCompile("[a-z]")
+	upperCase := regexp.MustCompile("[A-Z]")
+	numeric := regexp.MustCompile("[0-9]")
+	special := regexp.MustCompile("[^a-zA-Z0-9\\s]")
+
+	var redactInt = func(i int) int {
+		s := strconv.Itoa(i)
+		s = numeric.ReplaceAllString(s, "1")
+		i, _ = strconv.Atoi(s)
+		return i
+	}
+
+	switch b := b.(type) {
+	case map[string]any:
+		for k, v := range b {
+			if includedKeys[k] {
+				b[k] = v
+			} else {
+				b[k] = redactValues(v, includedKeys)
+			}
+		}
+		return b
+	case []any:
+		for i, v := range b {
+			b[i] = redactValues(v, includedKeys)
+		}
+		return b
+	case []uint8:
+		return "binary"
+	case string:
+		b = upperCase.ReplaceAllString(b, "A")
+		b = lowerCase.ReplaceAllString(b, "a")
+		b = numeric.ReplaceAllString(b, "1")
+		b = special.ReplaceAllString(b, "*")
```

Comment:
nit: The regex approach here is a little slower than you might expect (because you're recompiling the regexes on each call to the function, and iterating over the string creating a new string 4 times).

A more go-like way to do this might be:

```go
func redact(str string) string {
	redacted := ""
	for _, rune := range str {
		switch {
		case unicode.IsUpper(rune):
			redacted += "A"
		case unicode.IsLower(rune):
			redacted += "a"
		case unicode.IsNumber(rune):
			redacted += "1"
		case unicode.IsSpace(rune):
			redacted += " "
		default:
			redacted += "*"
		}
	}
	return redacted
}
```

Reply:
Oh shit - thank you